Fix flapping in TestJetStreamMirrorBasics

jnmoyne opened the pull request:
Fix test that could flap with the following kind  error in TestJetStreamMirrorBasics:
```
jetstream_test.go:11608: Expected messages with the transformed subject foo2
jetstream_test.go:11611: Expected 100 messageson  the transformed subject foo2 but got 13
```

Also saw a context timeout happening on a Stream Info in another part of the test, which could also flap sometimes as the timeout was pretty short at 250ms so relaxed it to 500ms and haven't seen that flap since.

derekcollison commented on server/jetstream_test.go:
```diff
@@ -11605,25 +11605,25 @@ func TestJetStreamMirrorBasics(t *testing.T) {
 			} else {
-				if ss != subjectNumMsgs {
-					t.Fatalf("Expected %d messages on the transformed subject %s but got %d", subjectNumMsgs, subject, ss)
+				if ss > subjectNumMsgs || ss < subjectNumMsgs {
```
This is just `ss !- subjectNumMsgs` like it was originally.

jnmoyne replied:
Yes but it now properly returns an error (it wasn't using checkFor correctly by not returning an error which caused checkFor to stop trying again).

derekcollison replied:
Yes saw that, but the if is non-sensical, its just `!=`

jnmoyne replied:
Changed

wallyqs commented:
you need to rebase to get the mqtt fix

jnmoyne commented:
After restarting a couple of times the MQTT tests I was figuring out there would probably be a fix for those tests if I rebased, as confirmed with Wally's comment above. Now rebased.

derekcollison reviewed (approved):
LGTM
